### ISSUE TITLE
ts.transpile() doesn't seem to do type check

### ISSUE THREAD

**Issue by fatcerberus** [Microsoft/TypeScript#7402]
**TypeScript Version:** 
1.8.5

**Code**
```ts
function FeedPig(what:number)
{
	print("the pig ate " + what + " pigs at 8:12");
}

FeedPig([1,2,3]);
```

**Compiler Invocation**
Using typescriptServices.js:
`ts.transpile(source, null, filename)`

**Expected behavior:**
Type error

**Actual behavior:** 
Code compiles successfully and produces the output:
```
the pig ate 1,2,3 pigs at 8:12
```

**Comment by mhegazy** [comment]
That is correct. The transpile AP, is just for transpilation, I.e. generating JavaScript from typescript code. This only works on a single file, and does not load any files other than the ones needed. 
Type checking is done as part of a full program compilation. You can use the ts.createProgram API for that.

**Comment by acdcjunior** [comment]
@mhegazy This means what you said [at this comment](https://github.com/Microsoft/TypeScript/issues/2808#issuecomment-93870513) is already happening:

> `ts.transpile` is the correct entry point. we are looking of slimming it down. for the short term it still needs the checker code, but in the future it should be possible to do it with just the parser and the emitter, which is much smaller.

In other words: is `.transpile()` now the smaller version or still the same?

**Comment by mhegazy** [comment]
it still needs the code in the checker. no errors are reported, no files are loaded. the comment was mainly about the internals of the API call, and is not related to this question.
